pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -329,6 +332,14 @@ func (am *Alertmanager) templateFromPaths(paths ...string) (*template.Template,
 	return tmpl, nil
 }
 
+func (am *Alertmanager) buildMuteTimesMap(muteTimeIntervals []apimodels.InternalMuteTimeInterval) map[string][]timeinterval.TimeInterval {
+	muteTimes := make(map[string][]timeinterval.TimeInterval, len(muteTimeIntervals))
+	for _, ti := range muteTimeIntervals {
+		muteTimes[ti.Name] = ti.TimeIntervals
```

Comment:
Is it possible that `muteTimeIntervals` contains two or more rows with the same `ti.Name`?

Reply:
Disclaimer: this code was taken from upstream Prometheus.

Technically speaking, it is possible, but the Name is used by the alertmanager as the unique identifier. So it's more of a misconfiguration where last once wins.